GoogleForCreators/web-stories-wp
# Media: Improve video captions appearance in editor

## Context



We recently improved how captions are displayed on the frontend.

Now we want to do the same in the editor as well.

## Summary



Adapts video captions appearance in the editor so that it matches the frontend experience, using a custom renderer for the video tracks.

## Relevant Technical Choices



* Adapted the logic from `amp-story-video-captions` to parse timestamps and manually render cues.
* Uses `dangerouslySetInnerHTML` to render the result of [`VTTCue.getCueAsHTML()`](https://developer.mozilla.org/en-US/docs/Web/API/VTTCue/getCueAsHTML)

## To-do



* Look into issues with captions not being displayed at all.
* Manual testing.

## User-facing changes


![Screenshot 2021-11-25 at 13 07 26](https://user-images.githubusercontent.com/841956/143438987-5df36bb2-1186-43c5-bbb1-4ecc7c8c0f53.png)

## Testing Instructions






This PR can be tested by following these steps:

1. Enable editor video captions experiment
2. Add a new video to a story and add captions to it
3. Move it around and maybe add some other elements on top of it so it's only partially visible
4. Play the video in the editor
5. See that the captions are displayed on top of all other elements and always at the same position on the canvas


## Reviews

### Does this PR have a security-related impact?



Using `dangerouslySetInnerHTML` in a couple of places, but the track contents should be considered trusted.

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #9344
